core/templates/dev/head/domain/exploration/SolutionObjectFactory.js

```diff
@@ -0,0 +1,125 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of Solution
+ * domain objects.
+ */
+
+oppia.factory('SolutionObjectFactory', [
+  '$filter', 'oppiaHtmlEscaper',
+  function($filter, oppiaHtmlEscaper) {
+    var Solution = function(answerIsExclusive, correctAnswer, explanation) {
+      this.answerIsExclusive = answerIsExclusive;
+      this.correctAnswer = correctAnswer;
+      this.explanation = explanation;
+    };
+
+    Solution.prototype.toBackendDict = function() {
+      return {
+        answer_is_exclusive: this.answerIsExclusive,
+        correct_answer: this.correctAnswer,
+        explanation: this.explanation
+      };
+    };
+
+    Solution.createFromBackendDict = function(solutionBackendDict) {
+      return new Solution(
+        solutionBackendDict.answer_is_exclusive,
+        solutionBackendDict.correct_answer,
+        solutionBackendDict.explanation);
+    };
+
+    Solution.createNew = function(
+      answerIsExclusive, correctAnswer, explanation) {
+      return new Solution(answerIsExclusive, correctAnswer, explanation);
+    };
+
+    Solution.prototype.getSolutionSummary = function(interactionId, choices) {
+      console.log(this.correctAnswer);
+      console.log(choices);
+      var isExclusiveAnswer = (
+        this.answerIsExclusive ? 'Only' : 'One');
+      var correctAnswer = '';
+      if (interactionId === 'GraphInput') {
```

Comment:
Typically for something with so many if statements you want to use a switch statement, makes the code easier to follow.